ServerCore/Pages/Teams/Hints.cshtml.cs

```diff
@@ -75,6 +75,15 @@ private async Task<IList<HintWithState>> GetAllHints(int puzzleID, int teamID)
                 {
                     hint.Discount = discount;
                 }
+
+                // if the event is over, show all hints
```

Comment:
To allow players to control hints they see afterward, it would be good to have something that special cases these for archiving